src/Compilers/CSharp/Test/Semantic/Semantics/SemanticAnalyzerTests.cs

```diff
@@ -720,13 +720,8 @@ void M()
                 Diagnostic(ErrorCode.ERR_MethGrpToNonDel, "M").WithArguments("M", "int").WithLocation(35, 18),
                 // (36,9): error CS1604: Cannot assign to 'this' because it is read-only
                 //         this = null;
-                Diagnostic(ErrorCode.ERR_AssgReadonlyLocal, "this").WithArguments("this").WithLocation(36, 9),
```

Comment:
I also didn't understand why those errors are gone.

Reply:
We used to not make the node bad so the tree looked like the illegal assignment actually happened. I made it uniform with all other similar cases - if the assignment is impossible, report an error and make node bad to avoid more errors.

---
In reply to: [135400179](https://github.com/dotnet/roslyn/pull/21334#discussion_r135400179) [](ancestors = 135400179)